File: clients/python/sdk/api/v1/accounts/item/accounts_item_request_builder.py
from __future__ import annotations
from dataclasses import dataclass
from kiota_abstractions.get_path_parameters import get_path_parameters
from kiota_abstractions.method import Method
from kiota_abstractions.request_adapter import RequestAdapter
from kiota_abstractions.request_information import RequestInformation
from kiota_abstractions.request_option import RequestOption
from kiota_abstractions.response_handler import ResponseHandler
from kiota_abstractions.serialization import Parsable, ParsableFactory
from kiota_abstractions.utils import lazy_import
from typing import Any, Callable, Dict, List, Optional, Union

block_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.block.block_request_builder')
featured_tags_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.featured_tags.featured_tags_request_builder')
follow_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.follow.follow_request_builder')
followers_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.followers.followers_request_builder')
following_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.following.following_request_builder')
lists_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.lists.lists_request_builder')
mute_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.mute.mute_request_builder')
note_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.note.note_request_builder')
pin_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.pin.pin_request_builder')
remove_from_followers_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.remove_from_followers.remove_from_followers_request_builder')
statuses_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.statuses.statuses_request_builder')
unblock_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.unblock.unblock_request_builder')
unfollow_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.unfollow.unfollow_request_builder')
unmute_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.unmute.unmute_request_builder')
unpin_request_builder = lazy_import('mastodon_client_lib.api.v1.accounts.item.unpin.unpin_request_builder')
account = lazy_import('mastodon_client_lib.models.account')

class AccountsItemRequestBuilder():
    """
    Builds and executes requests for operations under /api/v1/accounts/{id}
    """
    @property
    def block(self) -> block_request_builder.BlockRequestBuilder:
        """
        The block property
        """
        return block_request_builder.BlockRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def featured_tags(self) -> featured_tags_request_builder.Featured_tagsRequestBuilder:
        """
        The featured_tags property
        """
        return featured_tags_request_builder.Featured_tagsRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def follow(self) -> follow_request_builder.FollowRequestBuilder:
        """
        The follow property
        """
        return follow_request_builder.FollowRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def followers(self) -> followers_request_builder.FollowersRequestBuilder:
        """
        The followers property
        """
        return followers_request_builder.FollowersRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def following(self) -> following_request_builder.FollowingRequestBuilder:
        """
        The following property
        """
        return following_request_builder.FollowingRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def lists(self) -> lists_request_builder.ListsRequestBuilder:
        """
        The lists property
        """
        return lists_request_builder.ListsRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def mute(self) -> mute_request_builder.MuteRequestBuilder:
        """
        The mute property
        """
        return mute_request_builder.MuteRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def note(self) -> note_request_builder.NoteRequestBuilder:
        """
        The note property
        """
        return note_request_builder.NoteRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def pin(self) -> pin_request_builder.PinRequestBuilder:
        """
        The pin property
        """
        return pin_request_builder.PinRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def remove_from_followers(self) -> remove_from_followers_request_builder.Remove_from_followersRequestBuilder:
        """
        The remove_from_followers property
        """
        return remove_from_followers_request_builder.Remove_from_followersRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def statuses(self) -> statuses_request_builder.StatusesRequestBuilder:
        """
        The statuses property
        """
        return statuses_request_builder.StatusesRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def unblock(self) -> unblock_request_builder.UnblockRequestBuilder:
        """
        The unblock property
        """
        return unblock_request_builder.UnblockRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def unfollow(self) -> unfollow_request_builder.UnfollowRequestBuilder:
        """
        The unfollow property
        """
        return unfollow_request_builder.UnfollowRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def unmute(self) -> unmute_request_builder.UnmuteRequestBuilder:
        """
        The unmute property
        """
        return unmute_request_builder.UnmuteRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def unpin(self) -> unpin_request_builder.UnpinRequestBuilder:
        """
        The unpin property
        """
        return unpin_request_builder.UnpinRequestBuilder(self.request_adapter, self.path_parameters)
    
    def __init__(self,request_adapter: RequestAdapter, path_parameters: Optional[Union[Dict[str, Any], str]] = None) -> None:
        """
        Instantiates a new AccountsItemRequestBuilder and sets the default values.
        Args:
            pathParameters: The raw url or the Url template parameters for the request.
            requestAdapter: The request adapter to use to execute the requests.
        """
        if path_parameters is None:
            raise Exception("path_parameters cannot be undefined")
        if request_adapter is None:
            raise Exception("request_adapter cannot be undefined")
        # Url template to use to build the URL for the current request builder
        self.url_template: str = "{+baseurl}/api/v1/accounts/{id}"

        url_tpl_params = get_path_parameters(path_parameters)
        self.path_parameters = url_tpl_params
        self.request_adapter = request_adapter
    
    async def get(self,request_configuration: Optional[AccountsItemRequestBuilderGetRequestConfiguration] = None) -> Optional[account.Account]:
        request_info = self.to_get_request_information(
            request_configuration
        )
        if not self.request_adapter:
            raise Exception("Http core is null") 
        return await self.request_adapter.send_async(request_info, account.Account, None)
    
    async def patch(self,body: Optional[account.Account] = None, request_configuration: Optional[AccountsItemRequestBuilderPatchRequestConfiguration] = None) -> Optional[account.Account]:
        if body is None:
            raise Exception("body cannot be undefined")
        request_info = self.to_patch_request_information(
            body, request_configuration
        )
        if not self.request_adapter:
            raise Exception("Http core is null") 
        return await self.request_adapter.send_async(request_info, account.Account, None)
    
    def to_get_request_information(self,request_configuration: Optional[AccountsItemRequestBuilderGetRequestConfiguration] = None) -> RequestInformation:
        request_info = RequestInformation()
        request_info.url_template = self.url_template
        request_info.path_parameters = self.path_parameters
        request_info.http_method = Method.GET
        request_info.headers["Accept"] = "application/json"
        if request_configuration:
            request_info.add_request_headers(request_configuration.headers)
            request_info.add_request_options(request_configuration.options)
        return request_info
    
    def to_patch_request_information(self,body: Optional[account.Account] = None, request_configuration: Optional[AccountsItemRequestBuilderPatchRequestConfiguration] = None) -> RequestInformation:
        if body is None:
            raise Exception("body cannot be undefined")
        request_info = RequestInformation()
        request_info.url_template = self.url_template
        request_info.path_parameters = self.path_parameters
        request_info.http_method = Method.PATCH
        request_info.headers["Accept"] = "application/json"
        if request_configuration:
            request_info.add_request_headers(request_configuration.headers)
            request_info.add_request_options(request_configuration.options)
        request_info.set_content_from_parsable(self.request_adapter, "application/json", body)
        return request_info
    
    @dataclass
    class AccountsItemRequestBuilderGetRequestConfiguration():
        """
        Configuration for the request such as headers, query parameters, and middleware options.
        """
        # Request headers
        headers: Optional[Dict[str, str]] = None

        # Request options
        options: Optional[List[RequestOption]] = None

    
    @dataclass
    class AccountsItemRequestBuilderPatchRequestConfiguration():
        """
        Configuration for the request such as headers, query parameters, and middleware options.
        """
        # Request headers
        headers: Optional[Dict[str, str]] = None

        # Request options
        options: Optional[List[RequestOption]] = None

    

File: clients/csharp/sdk/Api/V1/Accounts/Item/AccountsItemRequestBuilder.cs
using MastodonClientLib.Api.V1.Accounts.Item.Block;
using MastodonClientLib.Api.V1.Accounts.Item.Featured_tags;
using MastodonClientLib.Api.V1.Accounts.Item.Follow;
using MastodonClientLib.Api.V1.Accounts.Item.Followers;
using MastodonClientLib.Api.V1.Accounts.Item.Following;
using MastodonClientLib.Api.V1.Accounts.Item.Lists;
using MastodonClientLib.Api.V1.Accounts.Item.Mute;
using MastodonClientLib.Api.V1.Accounts.Item.Note;
using MastodonClientLib.Api.V1.Accounts.Item.Pin;
using MastodonClientLib.Api.V1.Accounts.Item.Remove_from_followers;
using MastodonClientLib.Api.V1.Accounts.Item.Statuses;
using MastodonClientLib.Api.V1.Accounts.Item.Unblock;
using MastodonClientLib.Api.V1.Accounts.Item.Unfollow;
using MastodonClientLib.Api.V1.Accounts.Item.Unmute;
using MastodonClientLib.Api.V1.Accounts.Item.Unpin;
using MastodonClientLib.Models;
using Microsoft.Kiota.Abstractions;
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading;
using System.Threading.Tasks;
namespace MastodonClientLib.Api.V1.Accounts.Item {
    /// <summary>
    /// Builds and executes requests for operations under \api\v1\accounts\{id}
    /// </summary>
    public class AccountsItemRequestBuilder {
        /// <summary>The block property</summary>
        public BlockRequestBuilder Block { get =>
            new BlockRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The featured_tags property</summary>
        public Featured_tagsRequestBuilder Featured_tags { get =>
            new Featured_tagsRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The follow property</summary>
        public FollowRequestBuilder Follow { get =>
            new FollowRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The followers property</summary>
        public FollowersRequestBuilder Followers { get =>
            new FollowersRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The following property</summary>
        public FollowingRequestBuilder Following { get =>
            new FollowingRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The lists property</summary>
        public ListsRequestBuilder Lists { get =>
            new ListsRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The mute property</summary>
        public MuteRequestBuilder Mute { get =>
            new MuteRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The note property</summary>
        public NoteRequestBuilder Note { get =>
            new NoteRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>Path parameters for the request</summary>
        private Dictionary<string, object> PathParameters { get; set; }
        /// <summary>The pin property</summary>
        public PinRequestBuilder Pin { get =>
            new PinRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The remove_from_followers property</summary>
        public Remove_from_followersRequestBuilder Remove_from_followers { get =>
            new Remove_from_followersRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The request adapter to use to execute the requests.</summary>
        private IRequestAdapter RequestAdapter { get; set; }
        /// <summary>The statuses property</summary>
        public StatusesRequestBuilder Statuses { get =>
            new StatusesRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The unblock property</summary>
        public UnblockRequestBuilder Unblock { get =>
            new UnblockRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The unfollow property</summary>
        public UnfollowRequestBuilder Unfollow { get =>
            new UnfollowRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The unmute property</summary>
        public UnmuteRequestBuilder Unmute { get =>
            new UnmuteRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>The unpin property</summary>
        public UnpinRequestBuilder Unpin { get =>
            new UnpinRequestBuilder(PathParameters, RequestAdapter);
        }
        /// <summary>Url template to use to build the URL for the current request builder</summary>
        private string UrlTemplate { get; set; }
        /// <summary>
        /// Instantiates a new AccountsItemRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="pathParameters">Path parameters for the request</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public AccountsItemRequestBuilder(Dictionary<string, object> pathParameters, IRequestAdapter requestAdapter) {
            _ = pathParameters ?? throw new ArgumentNullException(nameof(pathParameters));
            _ = requestAdapter ?? throw new ArgumentNullException(nameof(requestAdapter));
            UrlTemplate = "{+baseurl}/api/v1/accounts/{id}";
            var urlTplParams = new Dictionary<string, object>(pathParameters);
            PathParameters = urlTplParams;
            RequestAdapter = requestAdapter;
        }
        /// <summary>
        /// Instantiates a new AccountsItemRequestBuilder and sets the default values.
        /// </summary>
        /// <param name="rawUrl">The raw URL to use for the request builder.</param>
        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
        public AccountsItemRequestBuilder(string rawUrl, IRequestAdapter requestAdapter) {
            if(string.IsNullOrEmpty(rawUrl)) throw new ArgumentNullException(nameof(rawUrl));
            _ = requestAdapter ?? throw new ArgumentNullException(nameof(requestAdapter));
            UrlTemplate = "{+baseurl}/api/v1/accounts/{id}";
            var urlTplParams = new Dictionary<string, object>();
            if (!string.IsNullOrWhiteSpace(rawUrl)) urlTplParams.Add("request-raw-url", rawUrl);
            PathParameters = urlTplParams;
            RequestAdapter = requestAdapter;
        }
        /// <param name="cancellationToken">Cancellation token to use when cancelling requests</param>
        /// <param name="requestConfiguration">Configuration for the request such as headers, query parameters, and middleware options.</param>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public async Task<Account?> GetAsync(Action<AccountsItemRequestBuilderGetRequestConfiguration>? requestConfiguration = default, CancellationToken cancellationToken = default) {
#nullable restore
#else
        public async Task<Account> GetAsync(Action<AccountsItemRequestBuilderGetRequestConfiguration> requestConfiguration = default, CancellationToken cancellationToken = default) {
#endif
            var requestInfo = ToGetRequestInformation(requestConfiguration);
            return await RequestAdapter.SendAsync<Account>(requestInfo, Account.CreateFromDiscriminatorValue, default, cancellationToken);
        }
        /// <param name="body">The request body</param>
        /// <param name="cancellationToken">Cancellation token to use when cancelling requests</param>
        /// <param name="requestConfiguration">Configuration for the request such as headers, query parameters, and middleware options.</param>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public async Task<Account?> PatchAsync(AccountUpdate body, Action<AccountsItemRequestBuilderPatchRequestConfiguration>? requestConfiguration = default, CancellationToken cancellationToken = default) {
#nullable restore
#else
        public async Task<Account> PatchAsync(AccountUpdate body, Action<AccountsItemRequestBuilderPatchRequestConfiguration> requestConfiguration = default, CancellationToken cancellationToken = default) {
#endif
            _ = body ?? throw new ArgumentNullException(nameof(body));
            var requestInfo = ToPatchRequestInformation(body, requestConfiguration);
            return await RequestAdapter.SendAsync<Account>(requestInfo, Account.CreateFromDiscriminatorValue, default, cancellationToken);
        }
        /// <param name="requestConfiguration">Configuration for the request such as headers, query parameters, and middleware options.</param>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public RequestInformation ToGetRequestInformation(Action<AccountsItemRequestBuilderGetRequestConfiguration>? requestConfiguration = default) {
#nullable restore
#else
        public RequestInformation ToGetRequestInformation(Action<AccountsItemRequestBuilderGetRequestConfiguration> requestConfiguration = default) {
#endif
            var requestInfo = new RequestInformation {
                HttpMethod = Method.GET,
                UrlTemplate = UrlTemplate,
                PathParameters = PathParameters,
            };
            requestInfo.Headers.Add("Accept", "application/json");
            if (requestConfiguration != null) {
                var requestConfig = new AccountsItemRequestBuilderGetRequestConfiguration();
                requestConfiguration.Invoke(requestConfig);
                requestInfo.AddRequestOptions(requestConfig.Options);
                requestInfo.AddHeaders(requestConfig.Headers);
            }
            return requestInfo;
        }
        /// <param name="body">The request body</param>
        /// <param name="requestConfiguration">Configuration for the request such as headers, query parameters, and middleware options.</param>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public RequestInformation ToPatchRequestInformation(AccountUpdate body, Action<AccountsItemRequestBuilderPatchRequestConfiguration>? requestConfiguration = default) {
#nullable restore
#else
        public RequestInformation ToPatchRequestInformation(AccountUpdate body, Action<AccountsItemRequestBuilderPatchRequestConfiguration> requestConfiguration = default) {
#endif
            _ = body ?? throw new ArgumentNullException(nameof(body));
            var requestInfo = new RequestInformation {
                HttpMethod = Method.PATCH,
                UrlTemplate = UrlTemplate,
                PathParameters = PathParameters,
            };
            requestInfo.Headers.Add("Accept", "application/json");
            requestInfo.SetContentFromParsable(RequestAdapter, "application/json", body);
            if (requestConfiguration != null) {
                var requestConfig = new AccountsItemRequestBuilderPatchRequestConfiguration();
                requestConfiguration.Invoke(requestConfig);
                requestInfo.AddRequestOptions(requestConfig.Options);
                requestInfo.AddHeaders(requestConfig.Headers);
            }
            return requestInfo;
        }
        /// <summary>
        /// Configuration for the request such as headers, query parameters, and middleware options.
        /// </summary>
        public class AccountsItemRequestBuilderGetRequestConfiguration {
            /// <summary>Request headers</summary>
            public RequestHeaders Headers { get; set; }
            /// <summary>Request options</summary>
            public IList<IRequestOption> Options { get; set; }
            /// <summary>
            /// Instantiates a new AccountsItemRequestBuilderGetRequestConfiguration and sets the default values.
            /// </summary>
            public AccountsItemRequestBuilderGetRequestConfiguration() {
                Options = new List<IRequestOption>();
                Headers = new RequestHeaders();
            }
        }
        /// <summary>
        /// Configuration for the request such as headers, query parameters, and middleware options.
        /// </summary>
        public class AccountsItemRequestBuilderPatchRequestConfiguration {
            /// <summary>Request headers</summary>
            public RequestHeaders Headers { get; set; }
            /// <summary>Request options</summary>
            public IList<IRequestOption> Options { get; set; }
            /// <summary>
            /// Instantiates a new AccountsItemRequestBuilderPatchRequestConfiguration and sets the default values.
            /// </summary>
            public AccountsItemRequestBuilderPatchRequestConfiguration() {
                Options = new List<IRequestOption>();
                Headers = new RequestHeaders();
            }
        }
    }
}
